llvm_passes/CMakeLists.txt

```diff
@@ -97,6 +103,14 @@ add_library(LLVMHipPasses MODULE HipPasses.cpp
     HipLowerZeroLengthArrays.cpp HipSanityChecks.cpp HipLowerSwitch.cpp
     ${EXTRA_OBJS})
 
+# From https://stackoverflow.com/a/48177101
+if (APPLE)
+  target_link_libraries(LLVMHipDynMem "-undefined dynamic_lookup")
+  target_link_libraries(LLVMHipStripUsedIntrinsics "-undefined dynamic_lookup")
+  target_link_libraries(LLVMHipDefrost "-undefined dynamic_lookup")
+  target_link_libraries(LLVMHipPasses "-undefined dynamic_lookup")
```

Comment:
These should potentially be `set_target_properties(LLVMHipDynMem PROPERTIES LINK_FLAGS "-undefined dynamic_lookup)` etc.; I'll have to read up on it.

I'm not sure why this is necessary in the first place (I think it should be the default for `MODULE` libs)

Works for now.